packages/presentation/lib/app/widgets/tabbar_widget.dart

```diff
@@ -2,25 +2,40 @@ import 'package:flutter/material.dart';
 import 'package:presentation/utils/colors.dart';
 import 'package:presentation/utils/dimensions.dart';
 
-class TabBarWidget extends StatelessWidget {
-  final VoidCallback _pop;
+class TabBarWidget extends StatefulWidget {
+  final VoidCallback _goToHomePage;
+  final VoidCallback _goToLoginPage;
 
-  const TabBarWidget(this._pop, {super.key});
+  const TabBarWidget(this._goToHomePage, this._goToLoginPage, {super.key});
+
+  @override
+  State<TabBarWidget> createState() => _TabBarWidgetState();
+}
+
+class _TabBarWidgetState extends State<TabBarWidget> {
+  int _currentIndex = 0;
+
+  set currentIndex(int index) => setState(() {
+        if (_currentIndex != index) {
+          _currentIndex = index;
+        }
+      });
```

Comment:
good idea, but what if we are at the nested screen of the tab and want to go to the root?

Reply:
Fixed.